lcls_tools/common/image_processing/roi.py

```diff
@@ -0,0 +1,109 @@
+from abc import ABC, abstractmethod
+from typing import (
+    List,
+)
+import numpy as np
+from matplotlib import patches
+from pydantic import BaseModel, PositiveFloat, Field
+
+
+class ROI(BaseModel, ABC):
+    roi_type: str
+    center: List[PositiveFloat]
+
+    @abstractmethod
+    def crop_image(self, img, **kwargs) -> np.ndarray:
+        """crop image using ROI"""
+        pass
+
+    @abstractmethod
+    def get_patch(self):
+        pass
+
+
+class CircularROI(ROI):
```

Comment:
Would an elliptical ROI be used? It could be the base class for circular where the x/y radii are the same

Reply:
I talked with Nicole about it, I will implement a elliptical ROI class.

Reply:
I just pushed a new version of this file. I had to do some funny pydantic workarounds. Could you take a look please?